### PR TITLE
deadlock: create clients with the same grpc env

### PR DESCRIPTION
**Pull request by youjiali1995**
Signed-off-by: youjiali1995 <zlwgx1023@gmail.com>

###  What have you changed?
Use the same grpc env to create deadlock clients 

###  What is the type of the changes?
- Bugfix (a change which fixes an issue)

###  How is the PR tested?
Manual tests.

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?
No

###  Does this PR affect `tidb-ansible`?
No

### CODE REVIEW

**Comment by youjiali1995** [comment]
@MyonKeminta @AndreMouche PTAL

File: src/storage/lock_manager/client.rs [hunk 1]
```diff
@@ -14,6 +14,20 @@ pub type Callback = Box<dyn Fn(DeadlockResponse)>;
 const CQ_COUNT: usize = 1;
 const CLIENT_PREFIX: &str = "deadlock";
 
+/// Builds the `Environment` of the client. All clients should use the same instance.
+fn env() -> Arc<Environment> {
```
Comment (siddontang) [inline src/storage/lock_manager/client.rs]: can we use the same client with pd and raft? 

IMO, I don't like using a static env for gRPC.

**Comment by overvenus** [comment]
What else will happen besides creating more threads?

**Comment by youjiali1995** [comment]
> What else will happen besides creating more threads?

No other influence.

**Pull request merged**